pkg/files/file_helpers.go

```diff
@@ -45,34 +47,109 @@ func GetPermissions(fileMode os.FileMode) string {
 	return fmt.Sprintf("%#o", fileMode.Perm())
 }
 
+func FileMode(mode string) os.FileMode {
+	result, err := strconv.ParseInt(mode, 8, 32)
+	if err != nil {
+		return os.FileMode(permissions)
+	}
+
+	return os.FileMode(result)
+}
+
 // GenerateConfigVersion returns a unique config version for a set of files.
 // The config version is calculated by joining the file hashes together and generating a unique ID.
-func GenerateConfigVersion(fileSlice []*v1.File) string {
+func GenerateConfigVersion(fileSlice []*mpi.File) string {
 	var hashes string
 
-	slices.SortFunc(fileSlice, func(a, b *v1.File) int {
+	slices.SortFunc(fileSlice, func(a, b *mpi.File) int {
 		return cmp.Compare(a.GetFileMeta().GetName(), b.GetFileMeta().GetName())
 	})
 
 	for _, file := range fileSlice {
 		hashes += file.GetFileMeta().GetHash()
 	}
 
-	return uuid.Generate("%s", hashes)
+	return GenerateHash([]byte(hashes))
 }
 
-// GenerateFileHash returns the hash value of a file's contents.
-func GenerateFileHash(filePath string) (string, error) {
-	f, err := os.Open(filePath)
-	if err != nil {
-		return "", err
+func GenerateHash(b []byte) string {
+	return uuid.NewMD5(uuid.Nil, b).String()
+}
+
+// GenerateHashWithReadFile returns the content of a file and the hash of that file.
+func GenerateHashWithReadFile(filePath string) ([]byte, string, error) {
```

Comment:
Feels like this function is doing too much. Would it be better to just have a `ReadFile` function and call the `GenerateHash` function outside of it?

```
contents, err := ReadFile("...")
hash := GenerateHash(contents)
```

Reply:
Fixed